### PR TITLE
Add Back Url to the Apps Model

### PR DESCRIPTION
**Pull request by romer8**
- [x] Added apps_back_url field to the App model
- [x] Allow to have a custom back url for any appliaction
- [x] Run and passed test
- [x] Created migrations.

### CODE REVIEW

File: tethys_apps/base/handoff.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def handoff(
                     urlish = handler(request, **kwargs)
                     return redirect(urlish)
                 except TypeError as e:
-                    error["message"] = "HTTP 400 Bad Request: {0}. ".format(str(e))
+                    error[" message"] = "HTTP 400 Bad Request: {0}. ".format(str(e))
```
Comment (swainn) [inline tethys_apps/base/handoff.py:123]: Why is a space added to this? Is this accidental?

File: tests/unit_tests/test_tethys_services/test_backends/test_hydroshare.py [hunk 2]
```diff
@@ -54,17 +54,12 @@ def test_extra_data(self):
...
-            name="",
-            alias="",
-        )
-
         hydro_share_auth2_obj = HydroShareOAuth2()
 
         hydro_share_auth2_obj.set_expires_in_to = 100
 
         ret = hydro_share_auth2_obj.extra_data(
-            "user1", "0001-009", mock_response, mock_details
```
Comment (swainn) [inline tests/unit_tests/test_tethys_services/test_backends/test_hydroshare.py:67]: I'd prefer @sdc50's solution here. @romer8 can you change it to this?

Comment (romer8) [reply tests/unit_tests/test_tethys_services/test_backends/test_hydroshare.py]: I have implemented this solution in the latest commit: https://github.com/romer8/tethys/commit/009a83e5efa8a9af51ed334c7bf20a4559f43093

Comment (romer8) [reply tests/unit_tests/test_tethys_services/test_backends/test_hydroshare.py]: I have also merge the main branch for the latest change

**Review by swainn — COMMENTED** [review]
Approved

**Pull request merged**